nalgebra-sparse/Cargo.toml

```diff
@@ -24,20 +24,33 @@ io      = [ "pest", "pest_derive" ]
 slow-tests = []
 
 [dependencies]
+criterion = { version = "0.3", features = ["html_reports"] }
 nalgebra = { version="0.30", path = "../" }
 num-traits = { version = "0.2", default-features = false }
 proptest = { version = "1.0", optional = true }
 matrixcompare-core = { version = "0.1.0", optional = true }
 pest           = { version = "2", optional = true }
 pest_derive    = { version = "2", optional = true }
 serde = { version = "1.0", default-features = false, features = [ "derive" ], optional = true }
+itertools = "0.10"
 
 [dev-dependencies]
 itertools = "0.10"
 matrixcompare = { version = "0.3.0", features = [ "proptest-support" ] }
 nalgebra = { version="0.30", path = "../", features = ["compare"] }
 serde_json = "1.0"
 
+[[example]]
+name = "spmm"
+required-features = ["io"]
+path = "example/spmm.rs"
+
 [package.metadata.docs.rs]
 # Enable certain features when building docs for docs.rs
 features = [ "proptest-support", "compare" ]
+
+[profile.release]
+opt-level = 3 
+lto = "fat"
+codegen-units = 1
+panic = "abort"
```

Comment:
I think we should remove the optimization flags here (and *especially* `panic = abort`!).